File: python_tests/helpers/test_data.py
"""
Test data helper for E2E Communication Platform tests
Provides mock data for testing various scenarios
"""

class TestDataHelper:
    """Helper class for managing test data"""
    
    @staticmethod
    def get_mock_users():
        """Get mock user data for testing"""
        return [
            {
                'id': 'U001',
                'name': 'John Smith',
                'email': 'john.smith@company.com',
                'department': 'IT',
                'role': 'Administrator',
                'status': 'Active',
                'location': 'Hong Kong'
            },
            {
                'id': 'U002', 
                'name': 'Mary Johnson',
                'email': 'mary.johnson@company.com',
                'department': 'Operations',
                'role': 'Manager',
                'status': 'Active',
                'location': 'Kowloon'
            },
            {
                'id': 'U003',
                'name': 'David Chen',
                'email': 'david.chen@company.com',
                'department': 'Engineering',
                'role': 'Engineer',
                'status': 'Inactive',
                'location': 'New Territories'
            },
            {
                'id': 'U004',
                'name': 'Sarah Wilson',
                'email': 'sarah.wilson@company.com',
                'department': 'Customer Service',
                'role': 'Operator',
                'status': 'Active',
                'location': 'Hong Kong'
            }
        ]
    
    @staticmethod
    def get_mock_roles():
        """Get mock role data for testing"""
        return [
            {
                'name': 'Administrator',
                'description': 'Full system access and management privileges',
                'permissions': ['user_management', 'role_management', 'settings_management', 'system_administration']
            },
            {
                'name': 'Manager',
                'description': 'Department management and reporting access',
                'permissions': ['user_view', 'notification_management', 'reporting']
            },
            {
                'name': 'Engineer',
                'description': 'Technical operations and maintenance access',
                'permissions': ['outage_management', 'technical_notifications', 'system_monitoring']
            },
            {
                'name': 'Operator',
                'description': 'Basic operational access',
                'permissions': ['notification_view', 'basic_operations']
            }
        ]
    
    @staticmethod
    def get_mock_settings():
        """Get mock settings data for testing"""
        return {
            'communication': {
                'email_notifications': True,
                'sms_notifications': True,
                'push_notifications': False
            },
            'delays': {
                'planned_outage_delay': 60,  # minutes
                'unplanned_outage_delay': 15,  # minutes
                'emergency_delay': 5  # minutes
            },
            'extreme_weather': {
                'enabled': False,
                'temperature_threshold': 35,  # Celsius
                'wind_speed_threshold': 60  # km/h
            }
        }
    
    @staticmethod
    def get_departments():
        """Get list of available departments"""
        return ['IT', 'Operations', 'Engineering', 'Customer Service', 'Finance', 'HR']
    
    @staticmethod
    def get_locations():
        """Get list of available locations"""
        return ['Hong Kong', 'Kowloon', 'New Territories', 'Lantau Island']
    
    @staticmethod
    def get_user_statuses():
        """Get list of user status options"""
        return ['Active', 'Inactive', 'Pending', 'Suspended']
    
    @staticmethod
    def get_permissions():
        """Get list of available permissions"""
        return [
            'user_management',
            'role_management', 
            'settings_management',
            'system_administration',
            'user_view',
            'notification_management',
            'reporting',
            'outage_management',
            'technical_notifications',
            'system_monitoring',
            'notification_view',
            'basic_operations'
        ]
    
    @staticmethod
    def create_test_user(user_id=None, name=None, email=None, department=None, role=None, status='Active', location=None):
        """Create a test user with specified or random data"""
        import random
        import string
        
        if not user_id:
            user_id = f"U{random.randint(100, 999)}"
        
        if not name:
            first_names = ['John', 'Mary', 'David', 'Sarah', 'Michael', 'Lisa', 'James', 'Jennifer']
            last_names = ['Smith', 'Johnson', 'Chen', 'Wilson', 'Brown', 'Davis', 'Miller', 'Moore']
            name = f"{random.choice(first_names)} {random.choice(last_names)}"
        
        if not email:
            email = f"{name.lower().replace(' ', '.')}@company.com"
        
        if not department:
            department = random.choice(TestDataHelper.get_departments())
        
        if not role:
            roles = [role['name'] for role in TestDataHelper.get_mock_roles()]
            role = random.choice(roles)
        
        if not location:
            location = random.choice(TestDataHelper.get_locations())
        
        return {
            'id': user_id,
            'name': name,
            'email': email,
            'department': department,
            'role': role,
            'status': status,
            'location': location
        }
    
    @staticmethod
    def create_test_role(name=None, description=None, permissions=None):
        """Create a test role with specified or random data"""
        import random
        
        if not name:
            role_prefixes = ['Test', 'Demo', 'Sample', 'Mock']
            role_suffixes = ['Role', 'User', 'Admin', 'Manager']
            name = f"{random.choice(role_prefixes)} {random.choice(role_suffixes)}"
        
        if not description:
            description = f"Test role for {name} with limited permissions"
        
        if not permissions:
            all_permissions = TestDataHelper.get_permissions()
            # Select 2-4 random permissions
            permission_count = random.randint(2, 4)
            permissions = random.sample(all_permissions, permission_count)
        
        return {
            'name': name,
            'description': description,
            'permissions': permissions
        }

File: playwright_tests/test-data/test-data.js
/**
 * Test Data Helper for E2E Communication Platform Tests
 * Provides mock data for testing various scenarios
 * JavaScript port of python_tests/helpers/test_data.py
 */

class TestDataHelper {
  /**
   * Get mock user data for testing
   */
  static getMockUsers() {
    return [
      {
        id: 'U001',
        name: 'John Smith',
        email: 'john.smith@company.com',
        role: 'System Admin',
        status: 'Active'
      },
      {
        id: 'U002',
        name: 'Mary Johnson',
        email: 'mary.johnson@company.com',
        role: 'Account Manager',
        status: 'Active'
      },
      {
        id: 'U003',
        name: 'David Chen',
        email: 'david.chen@company.com',
        role: 'CIC Team',
        status: 'Inactive'
      },
      {
        id: 'U004',
        name: 'Sarah Wilson',
        email: 'sarah.wilson@company.com',
        role: 'Management',
        status: 'Active'
      }
    ];
  }

  /**
   * Get mock role data for testing
   */
  static getMockRoles() {
    return [
      {
        name: 'System Admin',
        description: 'Full system access and management privileges',
        permissions: ['user_management', 'role_management', 'settings_management', 'system_administration']
      },
      {
        name: 'CIC Team',
        description: 'Communication and incident coordination access',
        permissions: ['user_view', 'notification_management', 'reporting']
      },
      {
        name: 'Account Manager',
        description: 'Customer account management access',
        permissions: ['customer_management', 'notification_management']
      },
      {
        name: 'Management',
        description: 'Management level access',
        permissions: ['reporting', 'notification_view']
      }
    ];
  }

  /**
   * Get mock settings data for testing
   */
  static getMockSettings() {
    return {
      communication: {
        email_notifications: true,
        sms_notifications: true,
        push_notifications: false
      },
      delays: {
        planned_outage_delay: 60,  // minutes
        unplanned_outage_delay: 15,  // minutes
        emergency_delay: 5  // minutes
      },
      extreme_weather: {
        enabled: false,
        temperature_threshold: 35,  // Celsius
        wind_speed_threshold: 60  // km/h
      }
    };
  }

  /**
   * Get list of available departments
   */
  static getDepartments() {
    return ['IT', 'Operations', 'Engineering', 'Customer Service', 'Finance', 'HR'];
  }

  /**
   * Get list of available locations
   */
  static getLocations() {
    return ['Hong Kong', 'Kowloon', 'New Territories', 'Lantau Island'];
  }

  /**
   * Get list of user status options
   */
  static getUserStatuses() {
    return ['Active', 'Inactive', 'Pending', 'Suspended'];
  }

  /**
   * Get list of available permissions
   */
  static getPermissions() {
    return [
      'user_management',
      'role_management',
      'settings_management',
      'system_administration',
      'user_view',
      'notification_management',
      'reporting',
      'outage_management',
      'technical_notifications',
      'system_monitoring',
      'notification_view',
      'basic_operations'
    ];
  }

  /**
   * Create a test user with specified or random data
   * @param {Object} options - User options
   * @returns {Object} User object
   */
  static createTestUser({
    userId = null,
    name = null,
    email = null,
    role = null
  } = {}) {
    if (!userId) {
      userId = `U${Math.floor(Math.random() * 900) + 100}`;
    }

    if (!name) {
      const firstNames = ['John', 'Mary', 'David', 'Sarah', 'Michael', 'Lisa', 'James', 'Jennifer'];
      const lastNames = ['Smith', 'Johnson', 'Chen', 'Wilson', 'Brown', 'Davis', 'Miller', 'Moore'];
      const firstName = firstNames[Math.floor(Math.random() * firstNames.length)];
      const lastName = lastNames[Math.floor(Math.random() * lastNames.length)];
      name = `${firstName} ${lastName}`;
    }

    if (!email) {
      email = `${name.toLowerCase().replace(' ', '.')}@company.com`;
    }

    if (!role) {
      const roles = this.getMockRoles().map(r => r.name);
      role = roles[Math.floor(Math.random() * roles.length)];
    }

    // New users are always Active (no status parameter needed)
    return {
      id: userId,
      name,
      email,
      role
    };
  }

  /**
   * Create a test role with specified or random data
   * @param {Object} options - Role options
   * @returns {Object} Role object
   */
  static createTestRole({
    name = null,
    description = null,
    permissions = null
  } = {}) {
    if (!name) {
      const rolePrefixes = ['Test', 'Demo', 'Sample', 'Mock'];
      const roleSuffixes = ['Role', 'User', 'Admin', 'Manager'];
      const prefix = rolePrefixes[Math.floor(Math.random() * rolePrefixes.length)];
      const suffix = roleSuffixes[Math.floor(Math.random() * roleSuffixes.length)];
      name = `${prefix} ${suffix}`;
    }

    if (!description) {
      description = `Test role for ${name} with limited permissions`;
    }

    if (!permissions) {
      const allPermissions = this.getPermissions();
      // Select 2-4 random permissions
      const permissionCount = Math.floor(Math.random() * 3) + 2; // 2-4
      permissions = [];
      const shuffled = [...allPermissions].sort(() => 0.5 - Math.random());
      permissions = shuffled.slice(0, permissionCount);
    }

    return {
      name,
      description,
      permissions
    };
  }

  /**
   * Generate bulk user data for upload testing
   * @param {number} count - Number of users to generate
   * @returns {Array} Array of user objects
   */
  static generateBulkUsers(count = 10) {
    const users = [];
    for (let i = 0; i < count; i++) {
      users.push(this.createTestUser({ userId: `BULK${String(i + 1).padStart(3, '0')}` }));
    }
    return users;
  }

  /**
   * Convert user data to CSV format for bulk upload testing
   * @param {Array} users - Array of user objects
   * @returns {string} CSV string
   */
  static usersToCSV(users) {
    const headers = ['User ID', 'Name', 'Email', 'Department', 'Role', 'Status', 'Location'];
    const rows = users.map(user => [
      user.id,
      user.name,
      user.email,
      user.department,
      user.role,
      user.status,
      user.location
    ]);
    
    return [
      headers.join(','),
      ...rows.map(row => row.join(','))
    ].join('\n');
  }
}

module.exports = TestDataHelper;
